llvm/tools/llvm-ir2vec/llvm-ir2vec.cpp

```diff
@@ -727,7 +739,19 @@ int main(int argc, char **argv) {
     } else if (EmbeddingsSubCmd) {
       if (!FunctionName.empty()) {
         // Process single function
-        Function *F = M->getFunction(FunctionName);
+        const Function *F = [&]() -> const Function * {
+          if (auto *ExactMatch = M->getFunction(FunctionName))
+            return ExactMatch;
+
+          const auto Demangled = llvm::demangle(FunctionName);
```

Comment:
Isn't `FunctionName` already demangled here?

Reply:
Nit: make this a separate helper function that can be called by both IR2Vec and MIR2Vec?

Reply:
> Nit: make this a separate helper function that can be called by both IR2Vec and MIR2Vec?

Accepted this